### DISCUSSION TITLE
How to get subsequence item

### DISCUSSION THREAD

**Discussion by drewen1** [fo-dicom/fo-dicom#1811] [Q&A]
I am trying to edit subsequence item tag. For example:  Referenced Series Sequence -> Referenced Instance Sequence
![subsequence](https://github.com/fo-dicom/fo-dicom/assets/22739159/8546a26e-544d-4529-83de-3928669e666e)

Could someone guide me how to write the code to get this? Thanks.

**Comment by mrbean-bremen** [comment]
You can get the sequence by using `GetSequence` on a dataset or a dataset item, and the list of sequence items via `Items`:
```cs
var referencedSeriesItems = dataset.GetSequence(DicomTag.ReferencedSeriesSequence).Items;
```
Then you have to may iterate over the items and find the one you need, and use the same to get sub-sequences. Provided you need the first item:
```cs
var referencedInstanceItems = referencedSeriesItems[0].GetSequence(DicomTag.ReferencedInstanceSequence).Items;
```
Same here - find the needed item(s), and here you can read or write tags:
```cs
var sopClassUid = referencedInstanceItems[0].GetString(DicomTag.ReferencedSOPClassUID);
referencedInstanceItems[0].AddOrUpdate(DicomTag.ReferencedSOPInstanceUID, "1.2.3.4");
```
etc., you get the gist. 
Note that this is out of my head, so there may be typos.
